agent/daemon.go

```diff
@@ -193,29 +161,60 @@ func (da DaemonAgent) GetAddrUnspent(addr string) ([]*tx.UTXO, error) {
 	if _, ok := addresses[addr]; !ok {
 		err := da.importAddress(addr)
 		if err != nil {
-			return nil, fmt.Errorf("fail to import address: %s", err.Error())
+			return fmt.Errorf("fail to import address: %s", err.Error())
 		}
 		err = da.getAllWatchedAddress(true)
 		if err != nil {
-			return nil, fmt.Errorf("fail to update watched address after import: %s", err.Error())
+			return fmt.Errorf("fail to update watched address after import: %s", err.Error())
 		}
 	}
 
-	utxos, err := da.listUnspent(addr)
-	if err != nil {
-		return nil, fmt.Errorf("fail to list all utxos from address: %s", err.Error())
+	// no unspent tx, check if it is an empty address
+	if used, err := da.isAddressUsed(addr); err != nil {
+		return err
+	} else if !used {
+		return ErrNoTxForAddr
 	}
-	if len(utxos) > 0 {
-		return utxos, nil
+	return ErrNoUnspentTx
```

Comment:
```suggestion
	return nil
```
The `ErrNoUnspentTx` is useless now.

Reply:
The error `agent.ErrNoUnspentTx` in the switch case at about LINE 213 could also be removed.